src/main/java/eventprocessing/models/ReadingAggregator.java

```diff
@@ -0,0 +1,93 @@
+package eventprocessing.models;
+
+import com.google.common.collect.Iterables;
+import eventprocessing.fileservices.CSVFileService;
+
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReadingAggregator {
+    private CSVFileService csvFileService;
+    private Clock clock;
+    private List<Bucket> buckets;
+    private int duplicateCounter = 0;
+
+    public int getDuplicateCounter() {
+        return duplicateCounter;
+    }
+
+    public List<Bucket> getBuckets() {
+        return buckets;
+    }
+
+    public ReadingAggregator(CSVFileService csvFileService, Clock clock) {
+        this.csvFileService = csvFileService;
+        this.clock = clock;
+        this.buckets = new ArrayList<>();
+        createInitialBuckets();
+    }
+
+    public void process(SqsResponse sqsResponse) {
+        assignResponseToBucket(sqsResponse);
+        Bucket bucket = removeExpiredBucket();
+        if (bucket != null) {
+            csvFileService.write(bucket);
+            createNextBucket();
+        }
+    }
+
+    private Bucket getExpiredBucket() {
+        Bucket expiredBucket = Iterables.getFirst(buckets, null);
+        if(expiredBucket.isExpiredAtTime(clock.millis())) {
```

Comment:
Why not pass the whole `clock` to `isExpiredAt`? It'll add more type safety, because a `long` could be any number, not just a time.